src/payment/models/buy/__tests__/buy.controller.spec.ts

```diff
@@ -50,8 +88,86 @@ describe('BuyController', () => {
+      name: 'DFX AG',
+      street: 'Bahnhofstrasse',
+      number: '7',
+      zip: '6300',
+      city: 'Zug',
+      country: 'Schweiz',
+    });
+  });
+
+  it('should return BF if amount > 9000', async () => {
+    jest.spyOn(buyService, 'createBuy').mockResolvedValue(createDefaultBuy());
+    jest.spyOn(fiatService, 'getFiat').mockResolvedValue(createCustomFiat({ name: 'CHF' }));
+    jest.spyOn(countryService, 'getCountryWithSymbol').mockResolvedValue(createDefaultCountry());
+
+    await expect(
+      controller.createBuyWithPaymentInfo(createJwt(), createBuyPaymentInfoDto(10000)),
+    ).resolves.toMatchObject({
+      iban: 'LI52088110104693K000C',
+      bic: 'BFRILI22',
+    });
+  });
+
+  it('should return BF if currency = USD', async () => {
+    jest.spyOn(buyService, 'createBuy').mockResolvedValue(createDefaultBuy());
+    jest.spyOn(fiatService, 'getFiat').mockResolvedValue(createCustomFiat({ name: 'USD' }));
+    jest.spyOn(countryService, 'getCountryWithSymbol').mockResolvedValue(createDefaultCountry());
+
+    await expect(controller.createBuyWithPaymentInfo(createJwt(), createBuyPaymentInfoDto())).resolves.toMatchObject({
+      iban: 'LI51088110104693K000U',
+      bic: 'BFRILI22',
+    });
+  });
+
+  it('should return Olkypay if currency = EUR & sctInst', async () => {
+    jest.spyOn(buyService, 'createBuy').mockResolvedValue(createDefaultBuy());
+    jest.spyOn(fiatService, 'getFiat').mockResolvedValue(createCustomFiat({ name: 'EUR' }));
+    jest.spyOn(countryService, 'getCountryWithSymbol').mockResolvedValue(createDefaultCountry());
+
+    await expect(controller.createBuyWithPaymentInfo(createJwt(), createBuyPaymentInfoDto())).resolves.toMatchObject({
+      iban: 'LU116060002000005040',
+      bic: 'OLKILUL1',
+    });
+  });
+
+  it('should return MB if ibanCountry = MBCountry & userDataCountry = MBCountry', async () => {
+    jest.spyOn(buyService, 'createBuy').mockResolvedValue(createDefaultBuy());
+    jest.spyOn(fiatService, 'getFiat').mockResolvedValue(createCustomFiat({ name: 'CHF' }));
+    jest
+      .spyOn(countryService, 'getCountryWithSymbol')
+      .mockResolvedValue(createCustomCountry({ maerkiBaumannEnable: true }));
+
+    await expect(controller.createBuyWithPaymentInfo(createJwt(), createBuyPaymentInfoDto())).resolves.toMatchObject({
+      iban: 'CH3408573177975200001',
+      bic: 'MAEBCHZZ',
+    });
+  });
+
+  it('should return BF as default', async () => {
+    jest.spyOn(buyService, 'createBuy').mockResolvedValue(createDefaultBuy());
+    jest.spyOn(fiatService, 'getFiat').mockResolvedValue(createCustomFiat({ name: 'CHF' }));
+    jest
+      .spyOn(countryService, 'getCountryWithSymbol')
+      .mockResolvedValue(createCustomCountry({ maerkiBaumannEnable: false }));
+
+    await expect(controller.createBuyWithPaymentInfo(createJwt(), createBuyPaymentInfoDto())).resolves.toMatchObject({
+      iban: 'LI52088110104693K000C',
+      bic: 'BFRILI22',
+    });
+  });
```

Comment:
I would add a test with a special currency (e.g. GBP) to see if it falls back to the EUR account.